### ISSUE TITLE
Mirror's Edge Catalyst (1233570)

### ISSUE THREAD

**Issue by Sterophonick** [ValveSoftware/Proton#6452]
# Compatibility Report
- Mirror's Edge Catalyst
- AppID 1233570

## System Information
Steam Deck
Driver: Mesa 22.2.0 (git-17e5312102)

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 
Crashes on boot due to EA App. Did not occur on Origin.

## Reproduction
Boot the game.

## PROTON_LOG
[steam-1233570.log](https://github.com/ValveSoftware/Proton/files/10415754/steam-1233570.log)

**Comment by tommipontinen76** [comment]
Why is this closed. 

Also because the new EA App, the game is unplayable on linux

**Comment by asier-ochoa** [comment]
> Why is this closed.
> 
> Also because the new EA App, the game is unplayable on linux

Yep, I'm having this same issue, all EA App games are unplayable due to their launcher crashing on startup.

**Comment by Swizzler121** [comment]
One year later, any update? I'm confused as to why this game has the "playable" steam deck badge, on my steam deck the game hangs after the initial setup of the EA launcher and on my desktop the game crashes to the EA launcher when trying to set the contrast during initial boot setup...

**Comment by Swizzler121** [comment]
I saw a comment on protondb saying people were able to get the game working again by rolling back to proton 8, did the same and immediately was able to play with no problems on my PC and deck. There appears to be a regression in proton 9 that prevents it from launching. I launched it with stock proton on my desktop to see if I could find out what is tripping it up in proton 9, but I couldn't figure out anything specific, couldn't see a specific crash just a bunch of warnings, attached the log for those smarter than me
[steam-1233570.log](https://github.com/user-attachments/files/16260859/steam-1233570.log)

**Comment by marek-lach** [comment]
Can someone confirm, if using the latest Proton Experimental as of this date, fixes it for them?

**Comment by ipr** [comment]
> Can someone confirm, if using the latest Proton Experimental as of this date, fixes it for them?

No, doesn't work. Maybe it is the "EA app" that is causing issues?

Log has following:
```
402021.203:012c:0130:err:combase:RoGetActivationFactory Failed to find library for L"Windows.Globalization.Calendar"
```
That sounds something might be specific to Windows/.net version?

Anyway, log attached.

[steam-1233570.log.gz](https://github.com/user-attachments/files/19544008/steam-1233570.log.gz)

**Comment by MkThUnderwd** [comment]
### GOOD NEWS
MEC is running on Proton 10 through Lutris.

### PROBLEM
Launching normally has the game crash before the "resume game" screen. Clicking launch within EA App will always lead to a crash. I've confirmed this multiple times.

### WORKAROUND
If I launch EA App in Lutris, then launching Task Manager in Lutris for the EA App prefix, and then doing:
File -> Run -> the MEC executable
... runs the game.

### RESULT
The game _does not crash_ when started outside of the EA App itself.
I've confirmed this too multiple times.
I haven't had any crashes during regular gameplay yet either.

### PROTON 9 & STEAM VERSIONS
This may work on Proton 9 too, but I did not test it.
I'm not sure how to do this fix on Steam and I have no way of testing without buying it there.
I would post to ProtonDB about it, but I do not own the game on Steam.

### SYSTEM SPECS
OS: Ubuntu 24.04.2 LTS
KERNEL: 6.11.0-26-generic
CPU: AMD Ryzen 7 5800XT 8-Core
GPU: AMD Radeon RX 6700 XT (radeonsi, navi22, LLVM 18.1.7, DRM 3.61, 6.11.0-26-generic)
GPU DRIVER: 4.6 Mesa 24.2.0-devel
RAM: 32 GB

### NETWORKING
I added 
`127.0.0.1	winter15.gosredirector.ea.com`
to the '/etc/hosts' file since the servers shut down.
I don't know if it's necessary. I had it that way in both test scenarios.

**Comment by kisak-valve** [comment]
Mirror's Edge Catalyst won't work on Linux Mint... but Mint is Ubuntu based... Seems to be the EA client that's the issue...

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9131.
@DiamondDoor posted on 2025-10-24T01:26:53:

Whenever I try to start mirror's edge, well, it used to let me get into the game start screen and the brightness adjustment screen, then it would just sit there and never let me actually start a game, and then it would crash and stop.

Now it doesn't even let me get that far. It immediately just goes into saying the game is running, even though it never lets me play the game at all, and then crashes and stops.

I think it's something to do with this idiotic EA client.

Is there any way around this?

Keep in mind that Final Fantasy 12 works just fine on this system. So Steam games will work on this system.

**Comment by DiamondDoor** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6452#issuecomment-2923346533

How do you launch the EA app in Lutris? How do you even download it for Linux (MINT in my case)?

**Comment by DiamondDoor** [comment]
A heads up for everyone interested... Since Valve and EA want to act like clowns instead of letting up play the game we paid for, you can wish list this game on GOG so hopefully one day we can play the game without this horrid EA app interfering. It's disgusting that I can pay full price for the game, sign up for the stupid EA app thing, jump through all these hoops, and still cannot play this game all because EA and Valve don't want to maintain access to this ridiculous EA app. If EA and Valve don't care about this game, then it's about time GOG did. So please cast your vote in the link below.

Here's the link https://www.gog.com/dreamlist/game/mirror-s-edge-catalyst-2016

**Comment by MkThUnderwd** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6452#issuecomment-3440267702

1. Proton 10.0 is what works for me for playing the game.

2. Install should work with 10.0, but I may have had to use a different one and switched later. 

I've tried many of them and even with Proton 10.0 the EA App was glitchy yet working the last time I tried.

If you get stuck somewhere in the install, close out programs in the prefix to shut it down and switch Proton versions in Lutris. This game has probably given me the worst time getting it to work.

Don't forget to switch back to Proton 10.0 and apply the fixes for server sun-setting.

Good luck

**Comment by Sterophonick** [comment]
Adding `127.0.0.1 winter15.gosredirector.ea.com` to your `/etc/hosts` works around the issue of it crashing on startup.

There's a catch, though. The game crashes in the initial load screen *unless* you use Proton 8.0. Something in WINE regressed that is leading to crashes on boot.

I've uploaded logs of Proton 8 vs Proton Experimental to help identify the regression.

[mec-proton8.log](https://github.com/user-attachments/files/23845611/mec-proton8.log)
[mec-protonexp.log](https://github.com/user-attachments/files/23845617/mec-protonexp.log)

**Comment by Patola** [comment]
This regression is still present in proton 10.0-3.
It works with Proton 8.0-5 but doesn't really need the /etc/hosts changes, as it just saves on dialog click -- game still works without it.
For completeness, I have generated 4 proton logs:
* Proton 8, without changes in /etc/hosts
* Proton 8 with altered /etc/hosts
* Proton 10, without changes in /etc/hosts
* Proton 10 with altered /etc/hosts

They are named accordingly, hope they are useful.

[steam-1233570-proton8-althost.log](https://github.com/user-attachments/files/23922532/steam-1233570-proton8-althost.log)
[steam-1233570-proton8.log](https://github.com/user-attachments/files/23922530/steam-1233570-proton8.log)
[steam-1233570-proton10.log](https://github.com/user-attachments/files/23922531/steam-1233570-proton10.log)
[steam-1233570-proton10-althost.log](https://github.com/user-attachments/files/23922533/steam-1233570-proton10-althost.log)

**Comment by ThatOneRuffian** [comment]
I was able to get Mirror's Edge: Catalyst working on Proton 8.0-5 after facing “An error on our end caused the launch to fail. Try again a little later.” After receiving this error, the would game crash on launch from the EA Launcher without error. Installing the proper dependencies also stopped the game from crashing after clicking OK on EA's server shutdown notice.

<img width="608" height="257" alt="Image" src="https://github.com/user-attachments/assets/58120c50-4cd3-43ed-943a-1fba4c64cb02" />

Here’s what worked for me:
1. Clear your steam game’s compatdata (making sure to backup your saved games):
    `rm -rf ~/.steam/steam/steamapps/compatdata/1233570  # steam game id 1233570 is for Mirror's Edge Catalyst`

2. Set game's proton version to 8.0-5.

3. Launch the game to initialize new game prefix and setup the EA Launcher.
4. After receiving the error, exit the game making sure that it shows as closed on Steam.
5. Now, run the winetricks script below to install the required dependencies.
```
#!/usr/bin/env bash

export WINEPREFIX="$HOME/.steam/steam/steamapps/compatdata/1233570/pfx"

winetricks dotnet45        # .NET Framework 4.5
winetricks ole32           # OLE 32‑bit COM libraries
winetricks oleaut32        # OLE Automation libraries
winetricks msxml6          # MSXML 6.0 parser
winetricks d3dcompiler_47  # Direct3D shader compiler v47
winetricks d3dx9           # Legacy DirectX 9 runtime
winetricks ie7             # Internet Explorer 7
winetricks win7            # Make Wine report Windows 7
```

6. Run the game (I typically get a non-fatal error here, just hit OK).

<img width="322" height="155" alt="Image" src="https://github.com/user-attachments/assets/41ff6bee-9e20-42ea-80e7-cf3da3719101" />

7. Login to your EA account.

8. Play.

These settings also fixed Dead Space 3 and Mass Effect: Legendary Edition for me. Just update the WINEPREFIX var in the script to target other game paths.

Hopefully this can help someone. Seems to be missing some dependencies required for the updated EA Launcher?

**Comment by Patola** [comment]
>     5. Now, run the winetricks script below to install the required dependencies.

 Avoid using winetricks with proton. protontricks works better and has more functions, like finding the appid for a game (it works for non-steam games even). Also, you can do all verbs in one command line, e.g. `protontricks 1233570 dotnet45 ole32 oleaut32 msxml6 ...`

I think these are not needed, tho, since I was able to run the game (purging the prefix before) without them.

**Comment by ThatOneRuffian** [comment]
@Patola Thank you. That was my first go to, but I had numerous problems installing and using protontricks on my system, from outdated version not being able to read steam's latest index magic number (system package), permission issues with bwrap (flatpak/python), etc. Not sure if that's related to the issues I have been having.

Edit:
Adding system specs

OS: Ubuntu 24.04.3 LTS
Kernel: 6.14.0-37-generic
CPU: AMD Ryzen 9 5950X 16-Core
GPU: AMD Radeon RX 9070 XT (radeonsi, gfx1201, LLVM 20.1.8, DRM 3.64, 6.14.0-37-generic)
GPU DRIVER: 4.6 Mesa 25.3.0-devel
RAM: 64 GB

**Comment by Dwarf1er** [comment]
System specs:
OS: Arch Linux x86_64
Host: B450M DS3H (Default string-CF)
Kernel: Linux 6.19.11-arch1-1
CPU: AMD Ryzen 5 3600 (12) @ 4.21 GHz
GPU: AMD Radeon RX 6800 XT
Memory: 16 GiB

Network changes to /etc/hosts:
127.0.0.1 winter15.gosredirector.ea.com

Proton version:
Proton Experimental

Launch settings (I was seeing a lot of  Unhandled Xalia exception):
PROTON_USE_XALIA=0

I've tried with all the other Proton versions mentioned in this thread, I also tried all of the protontricks mentioned and to no avail. The current behaviour is that EA App installs and then launches the game, you get the message about being unable to reach the game servers, then you get the initial loading screen, then it crashes, goes back to EA App which also crashes.

I've attached my log file, but a section that caught my eye was this:

```plaintext
11444.157:0494:057c:warn:seh:handle_syscall_fault backtrace: --- Exception 0xc0000005 at 0x7f614c91b74d: /home/iuse/.local/share/Steam/steamapps/common/Proton - Experimental/files/bin/../lib/wine/x86_64-unix/ntdll.so + 0x2174d (NtQueryDirectoryFile + 0xd8d).
11444.157:0494:057c:warn:seh:dump_syscall_fault backtrace: __wine_syscall_dispatcher.
11444.157:0494:057c:warn:seh:dump_syscall_fault backtrace: returning to user mode ip=00006ffffff3e9d4 ret=c0000005
11446.105:0494:0498:err:ole:marshal_object Failed to create an IRpcStubBuffer from IPSFactory for {e89f726e-c4f4-4c19-bb19-b647d7fa8478} with error 0x80004002
11458.262:0494:0498:err:ole:marshal_object Failed to create an IRpcStubBuffer from IPSFactory for {e89f726e-c4f4-4c19-bb19-b647d7fa8478} with error 0x80004002
11470.711:0494:0498:err:ole:marshal_object Failed to create an IRpcStubBuffer from IPSFactory for {e89f726e-c4f4-4c19-bb19-b647d7fa8478} with error 0x80004002
11490.157:0494:0498:err:ole:marshal_object Failed to create an IRpcStubBuffer from IPSFactory for {e89f726e-c4f4-4c19-bb19-b647d7fa8478} with error 0x80004002
```

I'm not so sure what could be causing this, but I will assume that EA App is doing something that's unsupported by Proton currently which causes the crash? I have no idea, but the game refuses to get past the initial loading screen without crashing.

EDIT:
Testing with a "non-EA" copy through Proton Experimental has the game running just fine. This seems to suggest that it's definitely EA App that is causing this crash.

[steam-1233570.log](https://github.com/user-attachments/files/26617146/steam-1233570.log)

**Comment by Dwarf1er** [comment]
I got Mirror's Edge Catalyst to launch by doing the following:

1. Set the game to use Proton Experimental
2. Set the following launch options: `PROTON_DISABLE_NVAPI=1 PROTON_NO_ESYNC=1 PROTON_USE_XALIA=0 %command%`
3. Add this line to my `/etc/hosts` file: `127.0.0.1 winter15.gosredirector.ea.com`
4. When EA App install screen pops up instead of clicking "let's go" click on "customize setup" and uncheck the box "create desktop shortcut"

**Comment by sam-k0** [comment]
Just installed the game today and got it running thanks to everyone here, but I think I did even less steps:

I am underneath Linux Mint 22.3
I downloaded the game on steam and got EA crap installed using Proton 9 (if this doesnt work, just try with different versions until the setup doesnt fail anymore)
Then to launch the actual game, I use Proton Experimental and the launch args provided by @Dwarf1er :
`PROTON_DISABLE_NVAPI=1 PROTON_NO_ESYNC=1 PROTON_USE_XALIA=0 %command%`

I didnt have to change the hosts file at all. Launching it made it hang for a second but then I got the Error that the EA servers are offline, but I could just continue.

**Comment by ipr** [comment]
>  using Proton Experimental

Did not start with experimental, 11.0 beta or 10.0, but it did finally start with 9.0-4. The others just had a case where EA app installation breaks itself so it doesn't even attempt to start with those.

And interestingly with those options it didn't crash while loading but actually got into game this time. I wonder what has changed...

**Comment by codeweaverwill** [comment]
I have been unable to reproduce the EA app installation breakage on any machine. An AMD-specific bug was found involving NVAPI, so that got a workaround in commit a07fb73. If someone who can repro could give experimental bleeding edge a shot (after deleting/renaming the prefix) to see if it helps, I'd appreciate it.
